app/controllers/application_controller.rb

```diff
@@ -43,4 +45,15 @@ def destroy_user_if_current_user
     @user = User.new
   end 
   
+  private
+  
+  def extract_locale_from_accept_language_header
+    language = request.env['HTTP_ACCEPT_LANGUAGE'].presence
+    if language
+      locale = HttpAcceptLangParser.parse(language)
+      logger.debug "* Locale set using accept_language to '#{locale}'"
+      locale.in?(User::LANGUAGES.map(&:to_sym)) ? locale : nil
```

Comment:
The bad part is that `nil.to_sym` raises an error.

It's good do define and return things as symbols. Keep that. But when comparing, always convert both parts to string.

This rule only applies to strings/symbols - not integers or any other type